tiledb/sm/serialization/array.cc

```diff
@@ -217,6 +258,33 @@ Status array_from_capnp(
         make_shared<ArraySchema>(HERE(), array_schema_latest));
   }
 
+  // Deserialize array directory
+  if (array_reader.hasArrayDirectory()) {
+    const auto& array_directory_reader = array_reader.getArrayDirectory();
+    auto array_dir =
+        array_directory_from_capnp(array_directory_reader, array->array_uri());
+    array->set_array_directory(*array_dir);
+  }
+
+  if (array_reader.hasFragmentMetadataAll()) {
+    array->fragment_metadata().clear();
+    array->fragment_metadata().reserve(
+        array_reader.getFragmentMetadataAll().size());
+    for (auto frag_meta_reader : array_reader.getFragmentMetadataAll()) {
+      auto meta = make_shared<FragmentMetadata>(HERE());
+      // TODO: consider a new constructor for fragment meta or using the
+      // existing one
+      meta->set_storage_manager(storage_manager);
```

Comment:
I'd add them in fragment_metadata_from_capnp, reuse this comment in there as well, it will be fixed in the serialization refactor when we'll c41 construct the objects.